MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,72 +1,122 @@
 import UIKit
 
-final class MovieQuizViewController: UIViewController {
-    // MARK: - Lifecycle
+final class MovieQuizViewController: UIViewController, QuestionFactoryDelegate{
+    
+    @IBOutlet private var imageView: UIImageView!
+    @IBOutlet private var textLabel: UILabel!
+    @IBOutlet private var counterLabel: UILabel!
+    @IBOutlet private var yesButton: UIButton!
+    @IBOutlet private var noButton: UIButton!
+    
+    private var currentQuestionIndex: Int = 0
+    private var correctAnswer: Int = 0
+    private let questionsAmount: Int = 10
+    private var questionFactory: QuestionFactoryProtocol?
+    private var currentQuestion: QuizQuestion?
+    private var statisticService = StatisticServiceImplementation()
+    private var alertPresent: AlertPresenterProtocol?
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        print(NSHomeDirectory())
+        questionFactory = QuestionFactory(delegate: self)
+        questionFactory?.requestNextQuestion()
+        alertPresent = AlertPresenter(delegate: self)
+    }
+    
+    func didReceiveNextQuestion(question: QuizQuestion?) {
+        guard let question = question else{
+            return
+        }
+        currentQuestion = question
+        let viewModel = convert(model: question)
+        DispatchQueue.main.async {
+            [weak self] in self?.show(quiz: viewModel)
+        }
```

Comment:
список захвата и in принято оставлять на той же строке :)
```suggestion
        DispatchQueue.main.async { [weak self] in 
            self?.show(quiz: viewModel)
        }
```